doc/source/whatsnew/v2.0.0.rst

```diff
@@ -243,6 +243,9 @@ Copy-on-Write improvements
   a modification to the data happens) when constructing a Series from an existing
   Series with the default of ``copy=False`` (:issue:`50471`)
 
+- The :class:`DataFrame` constructor now will keep track of references when called
```

Comment:
IIRC, we don't tell users about how CoW is implemented under the hood.

Maybe we can say that the constructor will obey CoW rules when called with a DataFrame.

Reply:
Yeah this makes more sense, will change

Reply:
I would match this sentence with the bullet point above, since it is basically the same enhancement but for DataFrame instead of Series

Reply:
Done